src/core/meshcop/joiner.cpp

```diff
@@ -197,19 +201,31 @@ void Joiner::HandleDiscoverResult(otActiveScanResult *aResult)
         steeringData.SetLength(aResult->mSteeringData.mLength);
         memcpy(steeringData.GetValue(), aResult->mSteeringData.m8, steeringData.GetLength());
 
-        if (steeringData.DoesAllowAny() ||
-            (steeringData.GetBit(mCcitt % steeringData.GetNumBits()) &&
-             steeringData.GetBit(mAnsi % steeringData.GetNumBits())))
+        if (steeringData.DoesAllowAny())
         {
-            mJoinerUdpPort = aResult->mJoinerUdpPort;
-            mJoinerRouterPanId = aResult->mPanId;
-            mJoinerRouterChannel = aResult->mChannel;
-            memcpy(&mJoinerRouter, &aResult->mExtAddress, sizeof(mJoinerRouter));
+            VerifyOrExit(!mJoinerRouterIsSpecific);
+            VerifyOrExit(aResult->mLqi > mJoinerRouterLqi);
```

Comment:
Thread 1.1.1 Section 8.4.4.1 uses the term "best signal strength".  Why did you choose to compare on LQI vs. RSSI?

Reply:
I used LQI as it is is supposed to give a more complete picture of the signal strength than just receive power, such as taking SnR into account. However I now notice that throughout openthread RSSI is used. Is this because of the undefined (by 802.15.4) nature of how the LQI is generated, or is there some other reason? I am happy to make the change if you think it fits better

Reply:
I agree RSSI has its pros/cons.  At the same time, Thread 1.1.1 solely relies on using signal strength and link margin and avoids the use of IEEE 802.15.4's LQI primitive.  The IEEE 802.15.4 LQI is underspecified and many implementations do not provide good indication of SNR or  link margin, but rather the quality of decoding chips on a very small portion of the packet.  To stay consistent with Thread 1.1.1, I'd prefer if we stuck with signal strength for now.

Reply:
Thanks for the explanation, I just pushed the change